static/app/views/settings/components/dataSecrecy/index.tsx

```diff
@@ -0,0 +1,170 @@
+import {useEffect, useState} from 'react';
+import moment from 'moment-timezone';
+
+import {addErrorMessage, addSuccessMessage} from 'sentry/actionCreators/indicator';
+import BooleanField, {
+  type BooleanFieldProps,
+} from 'sentry/components/forms/fields/booleanField';
+import DateTimeField, {
+  type DateTimeFieldProps,
+} from 'sentry/components/forms/fields/dateTimeField';
+import Panel from 'sentry/components/panels/panel';
+import PanelAlert from 'sentry/components/panels/panelAlert';
+import PanelBody from 'sentry/components/panels/panelBody';
+import PanelHeader from 'sentry/components/panels/panelHeader';
+import {t, tct} from 'sentry/locale';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+
+type WaiverData = {
+  access_end: string | null;
+  access_start: string | null;
+};
+
+export default function DataSecrecy() {
+  const api = useApi();
+  const organization = useOrganization();
+
+  const [allowAccess, setAllowAccess] = useState(organization.allowSuperuserAccess);
+  const [allowDate, setAllowDate] = useState<WaiverData>();
+  const [allowDateFormData, setAllowDateFormData] = useState<string>('');
+
+  const {data, refetch} = useApiQuery<WaiverData>(
+    [`/organizations/${organization.slug}/data-secrecy/`],
+    {
+      staleTime: 3000,
+      retry: (failureCount, error) => failureCount < 3 && error.status !== 404,
```

Comment:
Why do we retry? I just don't see this often in our frontend code so wondering how this endpoint is different from others that needs retrying.

Reply:
most endpoints are able to use the default retry, but since the api will return a 404 if an org doesn't have a waiver currently, the default behavior of the api would be to keep refreshing then render a failure state. here we prevent this by not refreshing on 404s.